src/libostree/ostree-sign-ed25519.c

```diff
@@ -242,7 +250,17 @@ gboolean ostree_sign_ed25519_data_verify (OstreeSign *self,
         }
     }
 
-  return glnx_throw (error, "no valid ed25519 signatures found");
+  if (invalid_signatures)
+    {
+      g_assert_cmpuint (n_invalid_signatures, >, 0);
+      /* The test suite has a key ring with 100 keys.  This seems insane, let's
+       * cap a reasonable error message at 3.
+       */
+      if (n_invalid_signatures > 3)
+        return glnx_throw (error, "ed25519: Failed to validate, tried %u keys", n_invalid_signatures);
```

Comment:
How about

```suggestion
        return glnx_throw (error, "ed25519: Signature couldn't be verified; tried %u keys", n_invalid_signatures);
```

?

To make it more consistent with the `n<=3` case.

Reply:
I merged that and tweaked things further so we are less redundant in the multiple key case.